cypress/integration/console/devices/onboarding/qr-scan.spec.js

```diff
@@ -0,0 +1,231 @@
+    cy.findByText('End device type information').should('be.visible')
+    cy.findByLabelText('DevEUI').should('have.value', devEui).should('be.disabled')
+    cy.findByLabelText('Claim authentication code')
+      .should('have.value', device.cac)
+      .should('be.disabled')
+
+    cy.findByRole('button', { name: 'Add end device' }).click()
+    cy.wait('@claim-request')
+      .its('request.body')
+      .should('deep.equal', composeExpectedRequest(device))
+  })
+
+  it('succeeds scanning again', () => {
+    cy.findByTestId('full-error-view').should('not.exist')
+    cy.findByTestId('error-notification').should('not.exist')
+
+    // Open qr modal and scan.
+    cy.findByText('Scan end device code').should('be.visible').click()
+    cy.findByText('Scan end device QR code').should('be.visible')
+    cy.findByText('Please scan the QR code to continue.').should('exist')
+    cy.findByTestId('webcam-feed').should('be.visible')
+    cy.findByText('Found QR code data').should('be.visible')
+    cy.findByRole('button', { name: /Scan again/g }).click()
+    cy.findByTestId('webcam-feed').should('be.visible')
+  })
+
+  it('succeeds registering a device via a qr code', () => {
+    cy.findByTestId('full-error-view').should('not.exist')
+    cy.findByTestId('error-notification').should('not.exist')
+
+    // Open qr modal and scan.
+    cy.findByText('Scan end device code').should('be.visible').click()
+    cy.findByText('Scan end device QR code').should('be.visible')
+    cy.findByText('Please scan the QR code to continue.').should('exist')
+    cy.findByTestId('webcam-feed').should('be.visible')
+    cy.findByText('Found QR code data').should('be.visible')
+    cy.findByText('Apply').should('not.be.disabled').click()
+
+    // Display scanned data in form.
+    cy.findByText('QR code scanned successfully').should('be.visible')
+    cy.findByText('End device type information').should('be.visible')
+
+    cy.findByLabelText('End device brand')
+      .parents('.select__control')
+      .next('input')
+      .should('have.value', 'test-brand-otaa')
+
+    selectDevice({
+      brand_id: 'test-brand-otaa',
+      model_id: 'test-model3',
+      hw_version: '2.0',
+      fw_version: '1.0.1',
+      band_id: 'EU_863_870',
+    })
+
+    // End device registration.
+    cy.findByLabelText('Frequency plan').selectOption('EU_863_870_TTN')
+    cy.findByLabelText('JoinEUI').should('have.value', joinEui).should('be.disabled')
+    cy.findByLabelText('DevEUI').should('have.value', devEui).should('be.disabled')
+    cy.findByRole('button', { name: 'Reset' }).should('be.disabled')
+    cy.findByLabelText('AppKey').type(generateHexValue(32))
+    cy.findByLabelText('End device ID').should('have.value', `eui-${device.devEui}`)
+
+    cy.findByRole('button', { name: 'Add end device' }).click()
+
+    cy.location('pathname').should(
+      'eq',
+      `${Cypress.config('consoleRootPath')}/applications/${appId}/devices/eui-${device.devEui}`,
```

Comment:
Wouldn't it be good to wait until the device page was loaded and assert the created values in the device overview here?